### PR TITLE
Added Streams tutorial for RedisInsight v2.0

### PR DESCRIPTION
**Pull request by ajeetraina**

### CODE REVIEW

**Review by simonprickett — COMMENTED** [review]
@ajeetraina I don't see Streams supported yet in RedisInsight 2.0.4 preview which is the current one available for public download?

File: docs/explore/redisinsightv2/streams/index-streams.mdx [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+
+
+*   Create applications that publish and consume messages. Nothing extraordinary here, you could already do that with Redis Pub/Sub.
+*   Consume messages that are published even when the client application (consumer) is not running. This is a big difference from Redis Pub/Sub.
+*   Consume messages starting from a specific point. For example, read the whole history or only new messages. 
+
+In addition, Redis Streams has the concept of consumer groups. Redis Streams consumer groups, like the similar concept in [Apache Kafka](https://kafka.apache.org/), allows client applications to consume messages in a distributed fashion (multiple clients), making it easy to scale and create highly available systems.
+
+Let’s dive under the covers and see [Redis Streams](https://redis.io/topics/streams-intro) through the lens of RedisInsight. You will see how to use [Lettuce Java client](https://developer.redis.com/develop/java/#using-lettuce) to publish and consume messages using consumer groups.This is the first basic example that uses a single consumer.
```
Comment (simonprickett) [inline docs/explore/redisinsightv2/streams/index-streams.mdx]:
```suggestion
Let’s dive under the covers and see [Redis Streams](https://redis.io/topics/streams-intro) through the lens of RedisInsight. You will see how to use the [Lettuce Java client](https://developer.redis.com/develop/java/#using-lettuce) to publish and consume messages using consumer groups. This is the first basic example that uses a single consumer.
```

File: docs/explore/redisinsightv2/streams/index-streams.mdx [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+## Prerequisite:
```
Comment (simonprickett) [inline docs/explore/redisinsightv2/streams/index-streams.mdx]:
```suggestion
## Prerequisites:
```

File: docs/explore/redisinsightv2/streams/index-streams.mdx [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+-   [Install Apache Maven](https://maven.apache.org/install.html)
+-   [Install Redis](https://developer.redis.com/create)
+
+
+## Step 1: Download RedisInsight
+
+
+To install RedisInsight on your local system, you need to first download the software from the Redis website.
+
+[Click this link ](https://redis.com/redis-enterprise/redis-insight/#insight-form) to access a form that allows you to select the operating system of your choice.
```
Comment (simonprickett) [inline docs/explore/redisinsightv2/streams/index-streams.mdx]:
```suggestion
[Click this link](https://redis.com/redis-enterprise/redis-insight/#insight-form) to access a form that allows you to select the operating system of your choice.
```

File: docs/explore/redisinsightv2/streams/index-streams.mdx [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+![My Image](images/redisinsight_form.png)
```
Comment (simonprickett) [inline docs/explore/redisinsightv2/streams/index-streams.mdx]:
```suggestion
![ReidsInsight download form](images/redisinsight_form.png)
```

File: docs/explore/redisinsightv2/streams/index-streams.mdx [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+
+
+Run the installer. 
+
+Click "Add Database manually".
+
+![My Image](images/redisinsight_connection.png)
```
Comment (simonprickett) [inline docs/explore/redisinsightv2/streams/index-streams.mdx]:
```suggestion
![Connecting to a database with RedisInsight](images/redisinsight_connection.png)
```

File: docs/explore/redisinsightv2/streams/index-streams.mdx [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+  mvn clean verify
+ ```
+
+
+## Step 3. Run the producer(Post a new message)
+
+
+ ```bash
+
+  mvn exec:java -Dexec.mainClass="com.kanibl.redis.streams.simple.RedisStreams101Producer" -Dexec.args="5"
```
Comment (simonprickett) [inline docs/explore/redisinsightv2/streams/index-streams.mdx:72]: Is there a good reason why this example code isn't `com.redis.streams...`?

File: docs/explore/redisinsightv2/streams/index-streams.mdx [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+  [INFO] BUILD SUCCESS
+  [INFO] ------------------------------------------------------------------------
+  [INFO] Total time:  9.102 s
+  [INFO] Finished at: 2021-05-18T13:07:01Z
+  [INFO] ------------------------------------------------------------------------
+  
+ ```
+
+
+## Step 4. Run the consumer(Consume messages)
```
Comment (simonprickett) [inline docs/explore/redisinsightv2/streams/index-streams.mdx]:
```suggestion
## Step 4. Run the consumer (Consume messages)
```

File: docs/explore/redisinsightv2/streams/index-streams.mdx [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+
+In the first terminal, post 100 new messages:
+
+
+ ```bash
+  mvn exec:java -Dexec.mainClass="com.kanibl.redis.streams.simple.RedisStreams101Producer" -Dexec.args="100"
+ ```
+
+
+Let us try to visualise the stream key using the RedisInsight browser tool as shown below:
```
Comment (simonprickett) [inline docs/explore/redisinsightv2/streams/index-streams.mdx:123]: I don't think you can do this with the publically available version 2.0.4 of RedisInsight?

File: docs/explore/redisinsightv2/streams/index-streams.mdx [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+
+ ```bash
+  mvn exec:java -Dexec.mainClass="com.kanibl.redis.streams.simple.RedisStreams101Producer" -Dexec.args="100"
+ ```
+
+
+Let us try to visualise the stream key using the RedisInsight browser tool as shown below: 
+
+
+![alt_text](images/redisinsight_key.png "image_tooltip")
```
Comment (simonprickett) [inline docs/explore/redisinsightv2/streams/index-streams.mdx]:
```suggestion
![RedisInsight key manager](images/redisinsight_key.png "image_tooltip")
```

File: docs/explore/redisinsightv2/streams/index-streams.mdx [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+
+
+Open "Profiler" tool to analyse your Redis commands as shown below:
+
+![alt_text](images/redisinsight_profiler.png "image_tooltip")
```
Comment (simonprickett) [inline docs/explore/redisinsightv2/streams/index-streams.mdx]:
```suggestion
![RedisInsight profiler tool](images/redisinsight_profiler.png "image_tooltip")
```

File: docs/explore/redisinsightv2/streams/index-streams.mdx [hunk 11]
```diff
@@ -0,0 +1,144 @@
...
+
+
+## Additional Links
+
+- [Slowlog Configuration using RedisInsight](/explore/redisinsight/slowlog)
+- [Explore Redis keys using RedisInsight browser tool](/explore/redisinsight/browser)
+- [Memory Analysis using RedisInsight](/explore/redisinsight/memoryanalyzer)
+
```
Comment (simonprickett) [inline docs/explore/redisinsightv2/streams/index-streams.mdx]:
```suggestion
- [RU202 Redis Streams](https://university.redis.com/courses/ru202/) - a course at Redis University
```

**Comment by ajeetraina** [comment]
@simonprickett I just checked with RedisInsight team and it look like even it is just tagging, the full support is not available yet. Thanks for bringing this into my attention. We might have to wait for the support. Let us park this for a moment till we have complete support.

redisinsight_key

**Comment by simonprickett** [comment]
@ajeetraina sounds good - maybe swap this to a draft PR for the meantime?

**Comment by ajeetraina** [comment]
Closing this PR as of now as this feature is not supported.

**Pull request closed without merging**